feedservice/feedservice.go

```diff
@@ -0,0 +1,227 @@
+package feedservice
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/greenbone/eulabeia/connection"
+	"github.com/greenbone/eulabeia/feedservice/handler"
+	"github.com/greenbone/eulabeia/feedservice/redis"
+	"github.com/greenbone/eulabeia/models"
+)
+
+type feed struct {
+	mqtt    connection.PubSub
+	context string
+	rc      *redis.RedisConnection
+}
+
+// getSeverity returns a SeverityType with given tags
+func getSeverity(tags map[string]string) models.SeverityType {
+	var severetyVector string
+	var severetyVersion string
+	var severetyDate string
+
+	if v, ok := tags["severety_vector"]; ok {
+		severetyVector = v
+	} else {
+		severetyVector = tags["cvss_base_vector"]
+	}
+	if strings.Contains(severetyVector, "CVSS:3") {
+		severetyVersion = "cvss_base_v3"
+	} else {
+		severetyVersion = "cvss_base_v2"
+	}
+	if v, ok := tags["severety_date"]; ok {
+		severetyDate = v
+	} else {
+		severetyDate = tags["creation_date"]
+	}
+
+	return models.SeverityType{
+		SeverityVector:  severetyVector,
+		SeverityVersion: severetyVersion,
+		SeverityDate:    severetyDate,
+		SeverityOrigin:  tags["severity_origin"],
+	}
+}
+
+func getRefs(cves string, bids string, xrefs string) []models.RefType {
+	var cveSlice []string
+	var bidSlice []string
+	var xrefSlice []string
+
+	var l int
+
+	if cves != "" {
+		cveSlice = strings.Split(cves, ", ")
+		l += len(cveSlice)
+	}
+
+	if bids != "" {
+		bidSlice = strings.Split(bids, ", ")
+		l += len(bidSlice)
+	}
+
+	if xrefs != "" {
+		xrefSlice = strings.Split(xrefs, ", ")
+		l += len(xrefSlice)
+	}
+
+	ret := make([]models.RefType, l)
+	i := 0
+	for _, v := range cveSlice {
+		ret[i] = models.RefType{
+			Type: "CVE",
+			ID:   v,
+		}
+		i++
+	}
+	for _, v := range bidSlice {
+		ret[i] = models.RefType{
+			Type: "BID",
+			ID:   v,
+		}
+		i++
+	}
+	for _, v := range xrefSlice {
+		xref := strings.SplitN(v, ":", 2)
+		if len(xref) != 2 {
+			continue
+		}
+		ret[i] = models.RefType{
+			Type: xref[0],
+			ID:   xref[1],
+		}
+		i++
```

Comment:
you can get the range with 
```
for i, v :=range xrefSlicee {
```
then you don't need to create i before and i++ it.

Reply:
I need the `i` as a counter for `cveSlice`, `bidSlice` and `xrefSlice` as all three slices will be put in a single slice for references.

Reply:
oh, right my mistake.